### DISCUSSION TITLE
Cloudflare protocol optimisation causes slow downs

### DISCUSSION THREAD

**Discussion by TiredSysOp** [rr-/szurubooru#668] [Q&A]
I've found that using cloudflare's protocol optimisation causes szurubooru instances to behave strangely. Slow downs and becoming unresponsive, if multiple actions by a user are done whilst the page is loading and the user waits they'll eventually all happen at once.

Spent a couple days tampering with nginx and szurubooru thread counts but found the issue was with cloudflare's proxy and protocol optimisation. Anyone else have any experience with this?

Ultimately I turned them all off and everything has been grand

**Comment by pelicans45** [comment]
@TiredSysOp Which specific protocol optimization settings caused this behavior? "HTTP/2 to Origin", "HTTP/3 (with QUIC)", "0-RTT Connection Resumption"? Or have you not narrowed it down to which was responsible?

**Reply by TiredSysOp** [reply]
I've not narrowed it down, I'm playing with my public facing instance and didn't want to upset my users. I'll try to narrow it down when it's quiet.

What gave it away was slow loading and the "Origin is not allowed by Access-Control-Allow-Origin" error on occasion. I've only made an educated guess and turned off all the optimisation at cloudflare's end and got lucky.

There may be other variables but I've not tested them yet. Right now I can only assume but I've not done the legwork.

**Reply by pelicans45** [reply]
@TiredSysOp Any updates on this? I'm curious.
